zone/npc.cpp

```diff
@@ -1361,17 +1405,22 @@ uint32 ZoneDatabase::UpdateNPCTypeAppearance(Client* c, NPC* n)
 	return updated;
 }
 
-uint32 ZoneDatabase::DeleteSpawnLeaveInNPCTypeTable(const std::string& zone, Client* c, NPC* n)
+uint32 ZoneDatabase::DeleteSpawnLeaveInNPCTypeTable(const std::string& zone, Client* c, NPC* n, uint32 remove_spawngroup_flag)
 {
+	if (!n->respawn2) {
+		return 0;
+	}
+
 	const auto& l = Spawn2Repository::GetWhere(
 		*this,
 		fmt::format(
-			"`zone` = '{}' AND `spawngroupID` = {}",
+			"`id` = {} AND `zone` = '{}' AND `spawngroupID` = {}",
+			n->respawn2->GetID(),
 			zone,
 			n->GetSpawnGroupId()
 		)
 	);
-
+	
```

Comment:
Weird whitespace.

Reply:
Resolved in new [commit](https://github.com/EQEmu/Server/pull/4311/commits/6524e60ba154786a0f7a5727e0717644fbd6a1d3)